collection_mutation.cc

```diff
@@ -23,10 +23,10 @@
 
 #include "collection_mutation.hh"
 
-collection_mutation::collection_mutation(const collection_type_impl& type, collection_mutation_view v)
+collection_mutation::collection_mutation(const abstract_type& type, collection_mutation_view v)
     : _data(imr_object_type::make(data::cell::make_collection(v.data), &type.imr_state().lsa_migrator())) {}
```

Comment:
It's dangerous to leave it with such a midleading name.

I suggest to rename it to "aggregate_cell_mutation" or maybe uglier but even less ambiguous "multi_cell_mutation". Let's agree on a name now, but only do it as a follow up to avoid disturbing the series.

Reply:
Ah, I see it is discussed at some length below. But github will show my comment first!

Reply:
For this PR I'll leave the name as it is.